smack-experimental/src/main/java/org/jivesoftware/smackx/httpauthorizationrequest/element/ConfirmExtension.java

```diff
@@ -0,0 +1,119 @@
+/**
+ *
+ *  Copyright 2019-2023 Eng Chong Meng
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.httpauthorizationrequest.element;
+
+import javax.xml.namespace.QName;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+
+/**
+ * ExtensionElement <code>Conform</code> for HTTP Request.
+ * XEP-0070: Verifying HTTP Requests via XMPP (1.0.1 (2016-12-09))
+ */
+public class ConfirmExtension implements ExtensionElement {
+    public static final String ELEMENT = "confirm";
+    public static final String NAMESPACE = "http://jabber.org/protocol/http-auth";
+
+    public static final QName QNAME = new QName(NAMESPACE, ELEMENT);
+
+    public static final String ATTR_ID = "id";
+    public static final String ATTR_METHOD = "method";
+    public static final String ATTR_URL = "url";
+
+    private final String id;
+    private final String method;
+    private final String url;
```

Comment:
Can this variable be handled as a string all the ways. Otherwise need to do extract steps during xml phrase() and in toXML() method.
In aTalk app implementation, the value is used as string in prompting user to accept or reject the auth request.

Reply:
> The URI class is not intended to be in fields.

Care to elaborate on that?